src/test/java/org/openrewrite/kotlin/tree/ArrayTest.java

```diff
@@ -94,7 +94,7 @@ void conditionalArrayAccess() {
         );
     }
 
-    @ExpectedToFail("Fixed by PSI-based-parser")
+    @ExpectedToFail("array[0]++ is mapped to 9 Fake elements. Find a way to retrieve the correct FIR.")
```

Comment:
I don't know how often this will happen. Combinations of syntax sugar that are de-sugared will result in many fake elements. These types of scenarios would need to be mapped manually. We've added an exception in the `PsiElementAssocations` to reveal when we don't find the correct element.